src/deriv_api/DerivAPIBasic.js

```diff
@@ -185,11 +192,23 @@ export default class DerivAPIBasic extends DerivAPICalls {
         return this.subscription_manager.forgetAll(...types);
     }
 
+    keepAlivePing() {
+        this.ping({ ping: 1 });
+        this.reconnect_timeout = setTimeout(this.reconnect.bind(this), 5000);
+    }
+
+    pong() {
+        clearTimeout(this.reconnect_timeout);
+    }
+
     openHandler() {
         this.events.next({
             name: 'open',
         });
-
+        if (this.shouldReconnect) {
+            this.pong(); // clear previous timeouts
```

Comment:
At first `this.reconect_timeout` is empty string `''`, and here it will be cleared by `pong()`. but it is not a `timeout` handler, is it ok ?